ui/src/jobs/components/modal/DeleteJobModal.js

```diff
@@ -0,0 +1,69 @@
+import React, { useEffect, useRef, useState } from "react";
+import { addToast } from "@caraml-dev/ui-lib";
+import { useJobModal } from "./useJobModal";
+import { useTuringApi } from "../../../hooks/useTuringApi";
+import { ConfirmationModal } from "../../../components/confirmation_modal/ConfirmationModal";
+import { isEmpty } from "../../../utils/object";
+import { EuiFieldText } from "@elastic/eui";
+
+export const DeleteJobModal = ({
+  onSuccess,
+  deleteJobRef,
+}) => {
+  const closeModalRef = useRef();
+  const [deleteConfirmation, setDeleteConfirmation] = useState('')
+
+  const [job = {}, openModal, closeModal] = useJobModal(closeModalRef);
+
+  const [{ isLoading, isLoaded, error }, submitForm] =  useTuringApi(
+    `/projects/${job.project_id}/jobs/${job.id}`,
+    {
+      method: "DELETE",
+      headers: { "Content-Type": "application/json" },
+    },
+    {},
+    false
+  );
+
+  useEffect(() => {
+    if (!isEmpty(job)  && isLoaded && !error) {
+      addToast({
+        id: `submit-success-terminate-${job.id}`,
+        title: `Ensembling Jobs ${job.name} for ensembler ${job.ensembler_id} has been terminated!`,
+        color: "success",
+        iconType: "check",
+      });
+      onSuccess();
+      closeModal();
+    }
+  }, [isLoaded, error, job, onSuccess, closeModal]);
+
+  return (
+    <ConfirmationModal
+      title={["failed", "failed_submission", "failed_building", "completed"].includes(job.status) ? 'Delete Ensembling Jobs' : 'Terminate Ensembling Jobs'}
```

Comment:
Since this same check (on the status being `IN` or `NOT IN` this list of statuses) is repeated in multiple places, can we refactor this into a common function in `ui/src/services/job/JobStatus.js` and use that ? Like:
```javascript
const isActiveJobStatus = function(jobStatus) {
    return ["failed", "failed_submission", "failed_building", "completed"].includes(jobStatus);
}
```